streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
@guozhangwang it is the [JoinWindows](http://kafka.apache.org/0100/javadoc/org/apache/kafka/streams/kstream/JoinWindows.html) documentation.  Please let me know if you wish me to revert the last commit to restore the other semantics.